src/main/functions/DidSiopMetadata.ts

```diff
@@ -30,29 +32,50 @@ function getIntersection<T>(rpMetadata: Array<T> | T, opMetadata: Array<T> | T):
   return arrayA.filter((value) => arrayB.includes(value));
 }
 
-function verifySubjectIdentifiers(subjectIdentifiers: SubjectIdentifierType | SubjectIdentifierType[]): boolean {
-  if (subjectIdentifiers || subjectIdentifiers.length) {
-    if (Array.isArray(subjectIdentifiers)) {
-      return subjectIdentifiers.includes(SIOP.SubjectIdentifierType.DID);
+function verifySubjectIdentifiers(subjectSyntaxTypesSupported: string[]): boolean {
+  if (subjectSyntaxTypesSupported.length) {
+    if (Array.isArray(subjectSyntaxTypesSupported)) {
+      return subjectSyntaxTypesSupported.includes(SIOP.SubjectIdentifierType.DID);
     }
-    return subjectIdentifiers === SIOP.SubjectIdentifierType.DID;
-  } else {
-    return false;
   }
+  return false;
 }
 
 function supportedDidMethods(rpMethods: string[] | string, opMethods: string[] | string): Array<string> {
   const supportedDidMethods = getIntersection(rpMethods, opMethods);
-  if (!supportedDidMethods.length) {
+  if (!supportedDidMethods.length || (supportedDidMethods.length === 1 && supportedDidMethods[0] === SubjectIdentifierType.DID)) {
     throw Error(SIOPErrors.DID_METHODS_NOT_SUPORTED);
   }
   return supportedDidMethods;
 }
 
-function supportedCredentialsFormats(rpCredentials: string[] | string, opCredentials: string[] | string): Array<string> {
-  const supportedCredentials = getIntersection(rpCredentials, opCredentials);
+export function supportedCredentialsFormats(rpCredentials: Format, opCredentials: Format): Format {
+  if (!rpCredentials || !opCredentials || !Object.keys(rpCredentials).length || !Object.keys(opCredentials).length) {
+    throw new Error(SIOPErrors.CREDENTIALS_FORMATS_NOT_PROVIDED);
+  }
+  const supportedCredentials = getIntersection(Object.keys(rpCredentials), Object.keys(opCredentials));
   if (!supportedCredentials.length) {
     throw new Error(SIOPErrors.CREDENTIAL_FORMATS_NOT_SUPPORTED);
   }
-  return supportedCredentials;
+  const intersectionFormat: Format = {};
+  supportedCredentials.forEach(function (crFormat) {
+    const rpAlgs = [];
+    const opAlgs = [];
+    Object.keys(rpCredentials[crFormat]).forEach((k) => rpAlgs.push(...rpCredentials[crFormat][k]));
+    Object.keys(opCredentials[crFormat]).forEach((k) => opAlgs.push(...opCredentials[crFormat][k]));
+    let methodKeyRP = undefined;
+    let methodKeyOP = undefined;
+    Object.keys(rpCredentials[crFormat]).forEach((k) => (methodKeyRP = k));
+    Object.keys(opCredentials[crFormat]).forEach((k) => (methodKeyOP = k));
+    if (methodKeyRP !== methodKeyOP) {
+      throw new Error(SIOPErrors.CREDENTIAL_FORMATS_NOT_SUPPORTED);
+    }
+    const algs = getIntersection(rpAlgs, opAlgs);
+    if (!algs.length) {
+      throw new Error(SIOPErrors.CREDENTIAL_FORMATS_NOT_SUPPORTED);
+    }
+    intersectionFormat[crFormat] = {};
```

Comment:
Wouldn't it make sense to have the intersection method do most of this and have it return the more complex object?

Reply:
changed a little bit, but the point is the intersection method is generic and I don't want to mess up the other methods using it, so I created a wrapper for it